assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -240,6 +249,60 @@ const baseActions = {
 			type: REGISTER_MODULE,
 		};
 	},
+
+	/**
+	 * Sets settings component for a module.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} slug                 Module slug.
+	 * @param {WPElement} settingsComponent React component to render the settings panel. Default is the DefaultModuleSettings component.
+	 * @return {Object} Redux-style action.
+	 */
+	*setSettingsComponent( slug, settingsComponent ) {
```

Comment:
We shouldn't have this action, the settings component should be provided as part of `registerModule`.